library/src/main/java/org/apache/apex/malhar/lib/state/managed/Bucket.java

```diff
@@ -247,6 +253,9 @@ public DefaultBucket(long bucketId)
     public void setup(@NotNull ManagedStateContext managedStateContext)
     {
       this.managedStateContext = Preconditions.checkNotNull(managedStateContext, "managed state context");
+      if (!disableBloomFilter && bloomFilter == null) {
+        bloomFilter = new SliceBloomFilter(bloomFilterBitSize, 0.99);
+      }
```

Comment:
"bloomFilter" has to move to time bucket because bloomFilter under DefaultBucket will work only if the managedState is of type ManagedTimeUnifiedStateImpl. Here, DefaultBucket behaves as time Bucket.
If the managedState is of type ManagedStateImpl/ManagedTimeStateImpl then DefaultBucket behaves as key bucket.